template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/oozie/OozieRoleConfigProvider.java

```diff
@@ -25,18 +31,27 @@
 
     private static final String OOZIE_DATABASE_PASSWORD = "oozie_database_password";
 
+    private static final String OOZIE_CONFIG_SAFETY_VALVE = "oozie_config_safety_valve";
+
     @Override
     protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, TemplatePreparationObject source) {
         switch (roleType) {
             case OozieRoles.OOZIE_SERVER:
                 RdsView oozieRdsView = getRdsView(source);
-                return List.of(
-                        config(OOZIE_DATABASE_HOST, oozieRdsView.getHost()),
-                        config(OOZIE_DATABASE_NAME, oozieRdsView.getDatabaseName()),
-                        config(OOZIE_DATABASE_TYPE, oozieRdsView.getSubprotocol()),
-                        config(OOZIE_DATABASE_USER, oozieRdsView.getConnectionUserName()),
-                        config(OOZIE_DATABASE_PASSWORD, oozieRdsView.getConnectionPassword())
-                );
+                List<ApiClusterTemplateConfig> config = new ArrayList<>(List.of(
+                    config(OOZIE_DATABASE_HOST, oozieRdsView.getHost()),
+                    config(OOZIE_DATABASE_NAME, oozieRdsView.getDatabaseName()),
+                    config(OOZIE_DATABASE_TYPE, oozieRdsView.getSubprotocol()),
+                    config(OOZIE_DATABASE_USER, oozieRdsView.getConnectionUserName()),
+                    config(OOZIE_DATABASE_PASSWORD, oozieRdsView.getConnectionPassword())));
+                if (isOozieHA(source) && CloudPlatform.GCP == source.getCloudPlatform()) {
+                    // There is no load-balancer for GCP yet. If the first Oozie server is down, the second Oozie server can process actions.
+                    // But without the load-balancer the callbacks from Yarn will fail. By default the action check delay is 10 mins.
+                    // Reducing this to 2 mins.
+                    config.add(config(OOZIE_CONFIG_SAFETY_VALVE,
```

Comment:
Wait, what? I think in this case we should simply say that we don't support Oozie HA for GCP.

What exactly do you mean by _"There is no load-balancer for GCP yet. If the first Oozie server is down, the second Oozie server can process actions."_ ? If no LoadBalancer is configured then how will the request routed to the 2nd instance?

I don't think we should do this. This would still mean a 2 minute delay in processing which is unacceptable to a lot of customers.

Reply:
> If no LoadBalancer is configured then how will the request routed to the 2nd instance?

If there is no load-balancer there are only two internal clients that are affected. Hue and Yarn callback. External clients are unaffected because Knox does client side failover. For Yarn callback this is just a temporary hack to reduce the delay in a failure scenario. This is much better than having Oozie in just one server. This is not supported in GA mode. This is a temporary workaround to overcome the problem of having 1 CM template for all three cloud providers. We had an agreement in the design discussion that it is OK for GCP to be in this mode when there is no cloud provider-based load-balancer support. CC: @sidseth @shwethags @mschumpert 

2 mins are just swag and I am ok with even 1 second, if there are no huge performance penalties, in the normal mode of operation.

Reply:
Let's take this conversation to the Jira.

Reply:
After Jira conversation addressed this by using the originating server address for Yarn callbacks https://github.com/hortonworks/cloudbreak/pull/10747/files#diff-999ddd6303a84e8956ee01a9ebbf5dc3127d0948e9e1879e784474120edd9a05R50

Reply:
Follow-up question: if the template already contains a safety-valve, will this extend it, or override it? See: [7.2.11/cdp-data-engineering-ha.bp#L229-L234](https://github.com/hortonworks/cloudbreak/blob/d6d64bbb4ce24b40afa93290bb1a6e502a4e02be/core/src/main/resources/defaults/blueprints/7.2.11/cdp-data-engineering-ha.bp#L229-L234)

Reply:
It will extend it for different keys. For same keys the templates' value will be given preference, not the code.

Reply:
So the `oozie.service.HadoopAccessorService.nameNode.whitelist` property coming from the template will also remain in the safety-valve, correct? And this will not remove that property.

Reply:
Yes, that is correct. It will *not* remove that property.